lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -46,34 +47,101 @@ def compute(self, student, course_structure, scores_client, submissions_scores):
                 filter_func=possibly_scored,
                 start_node=self.location,
         ):
-            descendant = course_structure[descendant_key]
+            self._compute_block_score(student, descendant_key, course_structure, scores_client, submissions_scores)
 
-            if not getattr(descendant, 'has_score', False):
-                continue
+        self.all_total, self.graded_total = graders.aggregate_scores(self.scores, self.display_name, self.location)
 
+    def save(self, student, subsection, course):
+        """
+        Persist the SubsectionGrade.
+        """
+        visible_blocks = [
+            BlockRecord(unicode(location), weight, score.possible)
```

Comment:
Why does the `location` need to be serialized at this layer?  Shouldn't it remain as a `UsageKey` while in memory?